sklearn/linear_model/base.py

```diff
@@ -467,8 +467,7 @@ def fit(self, X, y, sample_weight=None):
         X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'],
                          y_numeric=True, multi_output=True)
 
-        if sample_weight is not None and np.atleast_1d(sample_weight).ndim > 1:
-            raise ValueError("Sample weights must be 1D array or scalar")
+        sample_weight = _check_sample_weight(sample_weight, X)
```

Comment:
Once `_check_sample_weight` is called, sample weight is never None, which means that `_rescale_data` is always called below.

I would rather put,
```py
if sample_weight is not None:
    sample_weight = _check_sample_weight(sample_weight, X)
```
here to keep previous behavior.